recipes/ncurses/all/conanfile.py

```diff
@@ -134,108 +137,116 @@ def _configure_autotools(self):
             "--without-profile",
             "--with-sp-funcs",
             "--disable-rpath",
-            "--datarootdir={}".format(tools.unix_path(os.path.join(self.package_folder, "res"))),
             "--disable-pc-files",
+            "--datarootdir=${prefix}/res",
         ]
         build = None
         host = None
         if self.settings.os == "Windows":
-            conf_args.extend([
+            tc.configure_args += [
                 "--disable-macros",
                 "--disable-termcap",
                 "--enable-database",
                 "--enable-sp-funcs",
                 "--enable-term-driver",
                 "--enable-interop",
-            ])
-        if self.settings.compiler == "Visual Studio":
-            build = host = "{}-w64-mingw32-msvc".format(self.settings.arch)
-            conf_args.extend([
+            ]
+        if is_msvc(self):
+            build = host = f"{self.settings.arch}-w64-mingw32-msvc"
+            tc.configure_args += [
                 "ac_cv_func_getopt=yes",
                 "ac_cv_func_setvbuf_reversed=no",
-            ])
-            autotools.cxx_flags.append("-EHsc")
-            if tools.Version(self.settings.compiler.version) >= 12:
-                autotools.flags.append("-FS")
```

Comment:
Re-adding this flag like this fixed the build for me locally (6.2, Debug, MDd, shared):
```python
if Version(self.settings.compiler.version) >= 12:
    tc.extra_cxxflags.append("-FS")
```

Reply:
Thanks, @Ahajha! Looks like this fixed the issue.